lib/motions/general-motions.coffee

```diff
@@ -379,11 +379,56 @@ class MoveToMiddleOfScreen extends MoveToScreenLine
     height = lastScreenRow - firstScreenRow
     Math.floor(firstScreenRow + (height / 2))
 
+class ScrollKeepingCursor extends MoveToLine
+  previousFirstScreenRow: 0
+  currentFirstScreenRow: 0
+
+  constructor: (@editor, @vimState) ->
+    super(@editor, @vimState)
+
+  select: (count, options) ->
+    @scrollScreen(count)
+    super(count, options)
```

Comment:
So I just tried out your changes. I noticed that when in visual mode, the cursor's position is not preserved relative to the screen. I think you may need to call `@scrollScreen()` *after* calling `super` here:

```coffee
result = super
@scrollScreen(count)
result
```